Hi, i'm Damon. 

I'm currently 3rd year Bioinformatics Apprentice at Anglia Rusking University working at Sanger Wellcome Genome Campus with ToLA (Tree of Life Assembly Team).

My GitHub contributions may appear as either DLBPointon or as dp24.

[![Anurag's GitHub stats](https://github-readme-stats.vercel.app/api?username=DLBPointon&show_icons=true&theme=radical)](https://github.com/anuraghazra/github-readme-stats)

During my time at these institutes I have gained a significant amount of experiace in:

## HTML and Javascript

<details>
  
- In creating a number of full stack projects, such as [GRIT-realtime](https://github.com/DLBPointon/grt-v5) (for use at Sanger where it integrates with the local JIRA platform), university projects such as a [COVID tracker](https://github.com/DLBPointon/covid19-graphs-DLBPointon) which is still viewable [Here](https://dlbpointon.github.io/covid19-graphs-DLBPointon/) as portfolio evidence.

- In personal projects. I use a tool called [Obsidian](https://obsidian.md/) a second brain designed to powerful and customisable whilst making use of Markdown. In light of this I have written [QuickAdd-NCBI](), which adds taxonomic data from NCBI into a note, and contributed to a number of projects which extend the use of Obsidian such as [QuickAdd-Books](https://github.com/DLBPointon/script_googleBooks_quickAdd), Booksidian and [Obsidian-Plotly](https://github.com/DLBPointon/obsidian-plotly). In the future I plan to create a full plugin to integrate a wider breadth of the NCBI API into Obsidian, especially due to the large number of biologists and bioinformaticians that use the software.

- Javascript. Due to experience from the above I have al written a significant amount of JS as well as some small Javascript scripts for ToLQC which graph various statistics.

</details>
  
## Python

<details>

- I have used Python in almost every project I have been involved with during my time at ARU and Sanger. The first being [gEVAL_cleaner](https://github.com/DLBPointon/gEVAL_cleaner), this script would download cdna, cds, rna and protein data of a given organism and prepare it for use in genomic alignment. 

- Some components of [GRIT-realtime](https://github.com/DLBPointon/grt-v5) were also written in Python, namely the script which pulls data from JIRA and massaged it into the format required for database uploading

- I have also written a number of "bots", simple reporting scripts most of which pull various data from JIRA and post them to Slack channels. BTK-announcer  relays ticket information on BlobTookKit requests to those involved with decontamination, Priority-list relays information on high priotity tickets to all, Weekly-report reports on what tickets have changed hands over the course of the past week for management.

</details>

## Pipelines

<details>

- Snakemake. For a universioty project, I wrote a variant annotation pipeline using Illumina data.

- Nextflow DSL2. I am currently working on translating the Snakemake annotation pipeline into Nextflow for wider adoption at Sanger in the DTOL project, This will be using the nf-core production guidelines, allowing for "plug and play" functionality.

- I have also aided in the building of TeloSearch and TreeVal (NF-core standard), both of which are DSL2 pipelines in use at Sanger.
  
</details>
