<div id="header" align="center">
  <img src="https://publish-01.obsidian.md/access/e92f913c6458c80f98557a83c6402d17/Images/logo4.png" width="220"/>
</div>

<div id="badges" align="center">
  <a href="www.linkedin.com/in/dlbpointon">
    <img src="https://img.shields.io/badge/LinkedIn-blue?style=for-the-badge&logo=linkedin&logoColor=white" alt="LinkedIn Badge"/>
  </a>
  <a href="https://twitter.com/DLBPointon">
    <img src="https://img.shields.io/badge/Twitter-blue?style=for-the-badge&logo=twitter&logoColor=white" alt="Twitter Badge"/>
  </a>
</div>

<div id="views" align="center">
  <img src="https://komarev.com/ghpvc/?username=DLBPointon&style=flat-square&color=blue" alt=""/>
</div>

<h1>
  Hi, I'm Damon
  <img src="https://media.giphy.com/media/hvRJCLFzcasrR4ia7z/giphy.gif" width="30px"/>
</h1>

---
### :man_technologist: About Me:

I'm a Bioinformatian working at Sanger Wellcome Genome Campus (UK) inside ToLA (Tree of Life Assembly Team). My day-to-day role includes building Nextflow Pipelines, namely [TreeVal](https://github.com/sanger-tol/treeval), for the scaling of genomic analysis.


### :fire: My Stats :
My GitHub contributions may appear as either DLBPointon or as dp24.

[![GitHub Streak](http://github-readme-streak-stats.herokuapp.com?user=DLBPointon&theme=dark&date_format=M%20j%5B%2C%20Y%5D&mode=weekly&type=png)](https://git.io/streak-stats)

[![Top Langs](https://github-readme-stats.vercel.app/api/top-langs/?username=DLBPointon&hide_progress=true&theme=dark#gh-dark-mode-only)](https://github.com/anuraghazra/github-readme-stats)


During my time at these institutes I have gained a significant amount of experience in:

### HTML and Javascript

<details>
  
- In creating a number of full stack projects, such as [GRIT-realtime](https://github.com/DLBPointon/grt-v5) (for use at Sanger where it integrates with the local JIRA platform), university projects such as a [COVID tracker](https://github.com/DLBPointon/covid19-graphs-DLBPointon) which is still viewable [Here](https://dlbpointon.github.io/covid19-graphs-DLBPointon/) as portfolio evidence.

- In personal projects. I use a tool called [Obsidian](https://obsidian.md/) a second brain designed to powerful and customisable whilst making use of Markdown. In light of this I have written [QuickAdd-NCBI](), which adds taxonomic data from NCBI into a note, and contributed to a number of projects which extend the use of Obsidian such as [QuickAdd-Books](https://github.com/DLBPointon/script_googleBooks_quickAdd), Booksidian and [Obsidian-Plotly](https://github.com/DLBPointon/obsidian-plotly). In the future I plan to create a full plugin to integrate a wider breadth of the NCBI API into Obsidian, especially due to the large number of biologists and bioinformaticians that use the software.

- Javascript. Due to experience from the above I have al written a significant amount of JS as well as some small Javascript scripts for ToLQC which graph various statistics.

</details>
  
### Python

<details>

- I have used Python in almost every project I have been involved with during my time at ARU and Sanger. The first being [gEVAL_cleaner](https://github.com/DLBPointon/gEVAL_cleaner), this script would download cdna, cds, rna and protein data of a given organism and prepare it for use in genomic alignment. 

- Some components of [GRIT-realtime](https://github.com/DLBPointon/grt-v5) were also written in Python, namely the script which pulls data from JIRA and massaged it into the format required for database uploading

- I have also written a number of "bots", simple reporting scripts most of which pull various data from JIRA and post them to Slack channels. BTK-announcer  relays ticket information on BlobTookKit requests to those involved with decontamination, Priority-list relays information on high priotity tickets to all, Weekly-report reports on what tickets have changed hands over the course of the past week for management.
  
- There are also a number of smaller scripts, I haven't uploaded to GitHub. These include projects in gene expression over time, simple graphing and data parsing.

</details>

### Pipelines

<details>

- Snakemake. For a universioty project, I wrote a variant annotation pipeline using Illumina data.

- Nextflow DSL2. I am currently working on translating the Snakemake annotation pipeline into Nextflow for wider adoption at Sanger in the DTOL project, This will be using the nf-core production guidelines, allowing for "plug and play" functionality.

- I have also aided in the building of TeloSearch and TreeVal (NF-core standard), both of which are DSL2 pipelines in use at Sanger.
  
</details>

- :seedling: I am also currently learning; Rust in order to modernise some software in use at Sanger, Angular and it's integration with PostgREST/PostgreSQL for a personal project and delving deeper into JavaScript to build plugins for the Obsidian Community. 
